server/plugin.go

```diff
@@ -395,6 +400,36 @@ func (p *Plugin) handleBump(w http.ResponseWriter, r *http.Request) {
 	p.PostBotCustomDM(foreignUser, message, todoMessage, foreignIssueID)
 }
 
+// API endpoint to retrieve plugin configurations
+func (p *Plugin) handleConfig(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	requestMethod := r.Method
+
+	if requestMethod != http.MethodGet {
+		http.Error(w, "Invalid request method", http.StatusMethodNotAllowed)
+		return
+	}
+
+	if p.configuration != nil {
```

Comment:
The configuration may have in the future "secret" information that should not reach all the users. Could you pass here only the sidebar buttons related config?
I am all in for making this extensible in the future (create a struct like "client config" which now will only have the setting to hide the buttons, and a function "configToClientConfig" that just copy the relevant values), but at the moment it is not needed. Feel free to do it the way you prefer.